src/vst2/Vst2PluginInstance.h

```diff
@@ -85,13 +85,19 @@ class Vst2PluginInstance : public AudioEffectX
    int blockpos;
 
 public:
+   enum State {
+      UNINITIALIZED  = 0,
+      INITIALIZED    = 1,
+      DEAD           = 2,
+   };
+
    SurgeSynthesizer* _instance;
    VstEvent* _eventptr[MAX_EVENTS];
    char _eventbufferdata[EVENTBUFFER_SIZE];
 
    int events_this_block, events_processed;
-   bool initialized;
    void init();
+   unsigned int state;
```

Comment:
C and C++ do does very little to make it a real type (compare to for enums in Rust), except scoping. I think using some integer type (unsigned char, unsigned int) is actually better choice because then you at least know the size.

Reply:
Yeah that’s true at compile and assembly eject time but not true at parse and warn time. If you, say, switch on a typed enum you get warnings about missing conditions. 

Also the state isn’t an int. It is a state. There’s a type theoretic reason for it to be so!

And size sure. But we aren’t streaming this anywhere. And if we did we would cast it to the size we wanted.

Reply:
It is now with enum type.